packages/babel-parser/src/parser/expression.js

```diff
@@ -387,11 +395,34 @@ export default class ExpressionParser extends LValParser {
   // operator that has a lower precedence than the set it is parsing.
 
   parseExprOp(
-    left: N.Expression,
+    left: N.Expression | N.PrivateName,
     leftStartPos: number,
     leftStartLoc: Position,
     minPrec: number,
   ): N.Expression {
+    if (this.isPrivateName(left)) {
```

Comment:
We should forbid parenthesized private identifiers:
```js
(#a) in v
```

Reply:
It's already forbidden; there is a test at https://github.com/babel/babel/blob/7459038db8476e6a5159af012c5da26b342e1d21/packages/babel-parser/test/fixtures/experimental/private-in/private-in-parenthesized/input.js.

It throws in the check at line 1249.